build.gradle

```diff
@@ -52,6 +54,10 @@ allprojects {
     targetCompatibility = JavaVersion.VERSION_11
   }
 
+  tasks.withType(KotlinCompile) {
+    kotlinOptions.freeCompilerArgs += "-Xcontext-receivers"
```

Comment:
Is this safe? I mean, context receivers are just fancy parameters so there really shouldn't be any concern with unstable APIs, and it's all internal code that no one else is calling so there's no compatibility concern. So... okay?

Reply:
I had the same thought, wasn't sure if this would mess up the public API at all. It did kinda cover the use case I wanted which was exposing an extension method as essentially "protected" to implementations of an interface, but I'm also not so devoted to type safety that I wouldn't just expose it to all receivers.

If the build passes its prob fine

Reply:
spotless didnt like it which is the threshold for me trying to make it work so i got rid of it

cool feature tho